src/pages/repertoire/[song].tsx

```diff
@@ -0,0 +1,188 @@
+import { MusicalNoteIcon, XMarkIcon } from "@heroicons/react/24/solid"
+import SuggestionLink from "@/components/suggestion/song-information/suggestion-link"
+import Instrument from "@/components/suggestion/instrument"
+import { Database } from "@/types/database"
+import { useRouter } from "next/router"
+import ErrorPopup from "@/components/popups/error-popup"
+import { deleteDivision, insertDivision } from "@/services/suggestion.service"
+import Spinner from "@/components/utils/spinner"
+import { useSupabaseClient, useUser } from "@supabase/auth-helpers-react"
+import { PostgrestError } from "@supabase/supabase-js"
+
+interface SongProps {
+  song: Song
+}
+
+const SongPage = (props: SongProps) => {
+  const [song, setSong] = useState<Song>(props.song)
+  const [showSpinner, setShowSpinner] = useState<boolean>(false)
+  const [showConversionError, setShowConversionError] = useState<boolean>(false)
+  const [showUpdateError, setShowUpdateError] = useState<string>("")
+  const supabase = useSupabaseClient<Database>()
+  const router = useRouter()
+  const user = useUser()
+  const uid = user?.id
+  const isAdmin = user?.app_metadata.claims_admin
+
+  const updateSong = () => {
+    setShowSpinner(true)
+    getSong(supabase, song.id)
+      .then((response) => {
+        response.data
+          ? setSong(response.data as Song)
+          : setShowUpdateError("Failed to update the song.")
+      })
+      .catch(() => setShowUpdateError("Failed to update the song."))
+      .finally(() => setShowSpinner(false))
+  }
+
+  const handleDivisionOperation = (error: PostgrestError | null, message: string) => {
+    if (error) setShowUpdateError(message)
+    else updateSong()
+  }
+
+  const updateOrDeleteDivision = async (
+    exists: boolean,
+    divisionOperation: DivisionDatabaseOperation,
+    divisionLength: number
+  ) => {
+    if (exists && divisionLength == 1) {
+      setShowUpdateError("You're not allowed to remove yourself from this instrument.")
+    } else if (exists) {
+      deleteDivision(supabase, divisionOperation).then(({ error }) => {
+        handleDivisionOperation(error, "Failed to remove user from the instrument.")
+      })
+    } else {
+      insertDivision(supabase, divisionOperation).then(({ error }) => {
+        handleDivisionOperation(error, "Failed to add user to the instrument.")
+      })
+    }
+  }
+
+  const selectInstrument = (songInstrument: SongInstrument) => {
+    if (!uid) return
+
+    setShowSpinner(true)
+
+    const division: DivisionDatabaseOperation = {
+      musician: uid,
+      song_instrument_id: songInstrument.id
+    }
+
+    const exists = songInstrument.division.some(({ musician }) => musician.id === uid)
+    updateOrDeleteDivision(exists, division, songInstrument.division.length)
+      .catch(() => setShowUpdateError("Failed to update the instrument division."))
+      .finally(() => setShowSpinner(false))
+  }
+
+  const displayButton = (): boolean => isAdmin
```

Comment:
Is only used once in this component. Direct use of isAdmin gives the same result